src/ssl.c

```diff
@@ -15924,198 +15943,150 @@ int wolfSSL_set_compression(WOLFSSL* ssl)
             return NULL;
         }
 
-        /* return list of CAs sent from the server */
-        if (ssl->options.side == WOLFSSL_CLIENT_END) {
-            WOLF_STACK_OF(WOLFSSL_X509)* sk;
-
-            sk = wolfSSL_get_peer_cert_chain(ssl);
-            if (sk != NULL) {
-                WOLF_STACK_OF(WOLFSSL_X509_NAME)* ret;
-                WOLFSSL_X509* x509;
-
-                ret = wolfSSL_sk_X509_NAME_new(NULL);
-                do {
-                    x509 = wolfSSL_sk_X509_pop(sk);
-                    if (x509 != NULL) {
-                        if (wolfSSL_X509_get_isCA(x509)) {
-                            if (wolfSSL_sk_X509_NAME_push(ret,
-                                    wolfSSL_X509_get_subject_name(x509)) != 0) {
-                                WOLFSSL_MSG("Error pushing X509 name to stack");
-                                /* continue on to try other certificates and
-                                 * do not fail out here */
-                            }
-                        }
-                        wolfSSL_X509_free(x509);
-                    }
-                } while (x509 != NULL);
-                wolfSSL_sk_X509_free(sk);
-                return ret;
-            }
-            return NULL;
-        }
-        else {
-            /* currently only can be set in the CTX */
-            return ssl->ctx->ca_names;
-        }
+        return SSL_CA_NAMES(ssl);
     }
-#endif /* SESSION_CERTS */
 
-
-    #if defined(OPENSSL_ALL) || defined(OPENSSL_EXTRA) || \
-        defined(WOLFSSL_NGINX) || defined (WOLFSSL_HAPROXY)
-    /* registers client cert callback, called during handshake if server
-       requests client auth but user has not loaded client cert/key */
-    void wolfSSL_CTX_set_client_cert_cb(WOLFSSL_CTX *ctx, client_cert_cb cb)
+    #if !defined(NO_RSA) && !defined(NO_CERTS)
```

Comment:
It seems weird to me that this requires `!defined(NO_RSA)`. Are we sure that's necessary?

Reply:
Removed. Might not be necessary.